java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java

```diff
@@ -30,14 +30,22 @@ public class EsSinkConfig extends CommonSinkConfig {
     /** Optional, delimiter for generating id */
     private String delimiter;
 
+    private String username;
+
+    private String password;
+
     @JsonCreator
-    public EsSinkConfig(
-            @JsonProperty(value = "url") String url,
-            @JsonProperty(value = "index") String index,
-            @JsonProperty(value = "delimiter") String delimiter) {
+    public EsSink7Config(
+            @JsonProperty(value = "url", required = true) String url,
+            @JsonProperty(value = "index", required = true) String index,
+            @JsonProperty(value = "delimiter") String delimiter,
+            @JsonProperty(value = "username") String username,
```

Comment:
Putting the field in the constructor will make the field become required. Is this expected?

Reply:
I think this is the requirement for the field in json. url and index are required here. For other sink, the fields is ensured by `mapper.configure(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES, true)`.

Reply:
I believe they are optional fields but e2e also serves as docs in another way. We'd better demonstrate how to connect to a es with auth.

Reply:
They are optional. This e2e test connects to es with auth.

Reply:
In current code, can users create an es sink without setting the username and password? 

Previously putting a field in the constructor will make the field a required field, no matter whether we mark it with `required = true` or not. So in current code it is likely that user will fail to create the sink without setting the username and password.

Conventionally we don't put the optional field in the constructor and instead mark the optional field with `JsonProperty` annotation.

Reply:
Sorry for the late response. Seems github does not notify me.

Yes, they can. Integration test does this test and at least for now, the missing field will become null. I will check the conventional usage.